server/lib/realtime/rls/replication_poller.ex

```diff
@@ -0,0 +1,210 @@
+defmodule Realtime.RLS.ReplicationPoller do
+  use GenServer
+
+  require Logger
+
+  alias DBConnection.Backoff
+
+  alias Realtime.Adapters.Changes.{
+    DeletedRecord,
+    NewRecord,
+    UpdatedRecord
+  }
+
+  alias Realtime.SubscribersNotification
+  alias Realtime.RLS.Replications
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  @impl true
+  def init(opts) do
+    state = %{
+      backoff:
+        Backoff.new(
+          backoff_min: Keyword.fetch!(opts, :backoff_min),
+          backoff_max: Keyword.fetch!(opts, :backoff_max),
+          backoff_type: Keyword.fetch!(opts, :backoff_type)
+        ),
+      poll_interval: Keyword.fetch!(opts, :replication_poll_interval),
+      poll_ref: make_ref(),
+      publication: Keyword.fetch!(opts, :publication),
+      slot_name: Keyword.fetch!(opts, :slot_name),
+      temporary_slot: Keyword.fetch!(opts, :temporary_slot)
+    }
+
+    {:ok, state, {:continue, :prepare_replication}}
+  end
+
+  @impl true
+  def handle_continue(
+        :prepare_replication,
+        %{backoff: backoff, slot_name: slot_name, temporary_slot: temporary_slot} = state
+      ) do
+    try do
+      Replications.prepare_replication(slot_name, temporary_slot)
+    catch
```

Comment:
@abc3 i've had to catch db connection errors in order to allow realtime to recover due to network connection issues. without it, both `ReplicationPoller` and `SubscriptionManager` restarts too fast which shuts down the entire application. thought this was better then wrapping both processes under supervisor and tweaking max_* settings